eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxos.hs

```diff
@@ -288,11 +289,13 @@ babbageEvalScriptsTxInvalid = do
   {- utxoDel  = txBody ^. collateralInputsTxBodyL ◁ utxo -}
   let !(utxoKeep, utxoDel) = extractKeys (unUTxO utxo) (txBody ^. collateralInputsTxBodyL)
       UTxO collouts = collOuts txBody
-      collateralFees = collAdaBalance txBody utxoDel -- NEW to Babbage
+      -- The collateral balance shouldn't be negative at this point, otherwise
+      -- it would have failed in phase 1
```

Comment:
This isn't exactly correct because this is also phase 1 and also we do not short circuit on validation failures. So, it can be negative, but there will be phase 1 validation failure when that is the case. 

I am not quite sure if we should rephrase this comment or just remove it altogether.
```suggestion
```